src/gui/MainWindow.cc

```diff
@@ -1759,23 +1759,30 @@ void MainWindow::actionTreasureTrove()
   extensions.replaceInStrings(QRegularExpression("^"), "*.");
   const auto filter = QString("OpenSCAD Backups (%1)").arg(extensions.join(" "));
 
-  // Bug in QT: dirname is ignored for getOpenFileName
+#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
+  // QT5 just shows the most recent files in my place , getSaveFileName workaround
   QString filename = QFileDialog::getSaveFileName(this, "Open File", dirname, filter);
+#else
+  QString filename = QFileDialog::getOpenFileName(this, "Open File", dirname, filter);
+#endif
 
   if(filename.size() == 0) return; // dont proceed when cancelled
   
   tabManager->actionNew();
 
-  std::ostringstream stream; // compile restored design
+  QString result;
+  QTextStream stream(&result); // compile restored design
   stream << "/*\n"; // put into comments in case it crashes (again)			     
   std::ifstream file( filename.toStdString());
   if ( file ) {
-    stream << file.rdbuf();
+    std::string file_content;
+    std::getline(file, file_content, '\0');
+    stream << file_content.c_str();
     file.close();
   }
```

Comment:
Qt file reading (without error checking):
```c++
QFile file("data.txt");
QTextStream in(&file);
QString text;
in >> text;
file.close();
```

Reply:
Scratch that, we obviously have a "load file from path" function already at `TabManager::refreshDocument()`. So it would make even more sense to extract that core number of lines and use it for both cases.